pyccel/decorators.py

```diff
@@ -10,66 +10,135 @@
+    optimisations like function reordering or parallelisation.
 
-def pure(f):
+    Parameters
+    ----------
+    f : Function
+        The function to which the decorator is applied.
+
+    Returns
+    -------
+    Function
+        The unchanged function.
+    """
     return f
 
 def private(f):
+    """
+    Indicate that a function shouldn't be exposed in the Python interface.
+
+    Indicate that a function shouldn't be exposed in the Python interface. It
+    is translated and can be called by other functions in the module but will
+    not be wrapped.
+
+    Parameters
+    ----------
+    f : Function
+        The function to which the decorator is applied.
+
+    Returns
+    -------
+    Function
+        The unchanged function.
+    """
     return f
 
 def elemental(f):
+    """
+    Indicate that the function can also be used element-wise.
+
+    This decorator indicates that the function can be applied element-wise to
+    array arguments. The function operates on scalar inputs and the wrapper
+    will be provided for the annotatied arguments, but in the low-level code
+    the function can be called on arrays whose data type matches the arguments.
+
+    Parameters
+    ----------
+    f : Function
+        The function to which the decorator is applied.
+
+    Returns
+    -------
+    Function
+        The unchanged function.
+    """
     return f
 
 def inline(f):
-    """Indicates that function calls to this function should
-    print the function body directly"""
+    """
+    Indicate that the function should be inlined in the low-level code.
+
+    This decorator indicates that the function should be inlined where it is
+    called. Calls are replace with the function body.
+
+    Parameters
+    ----------
+    f : Function
+        The function to which the decorator is applied.
+
+    Returns
+    -------
+    Function
+        The unchanged function.
+    """
     return f
 
-def stack_array(f, *args):
+def stack_array(*args):
     """
-    Decorator indicates that all arrays mentioned as args should be stored
-    on the stack.
+    Indicate that arrays should be stored on the stack.
+
+    This decorator indicates that all arrays whose names appear among the
+    arguments should be stored on the stack.
 
     Parameters
     ----------
-    f : Function
-        The function to which the decorator is applied
-    args : list of str
-        A list containing the names of all arrays which should be stored on the stack
+    *args : str
+        The names of all arrays which should be stored on the stack.
+
+    Returns
+    -------
+    Function
+        The identity decorator which will be applied to the function.
     """
     def identity(f):
         return f
     return identity
 
-def allow_negative_index(f,*args):
+def allow_negative_index(*args):
     """
+    Indicate that arrays can be accessed with negative indexes.
+
     Decorator indicates that all arrays mentioned as args can be accessed with
```

Comment:
Maybe the word "This" is missing?
```suggestion
    This decorator indicates that all arrays mentioned as args can be accessed with
```
Or otherwise,
```suggestion
    Decorator indicating that all arrays mentioned as args can be accessed with
```